### PR TITLE
Create Budget App and add tests

### PR DESCRIPTION
**Pull request by Theophileaseh**
## In this project milestone, I implemented the following project requirements:
[Video Link](https://www.loom.com/share/436983f6d98b4e68a5e0f5367744e4b5)

- Splash screen
> A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
> The user should be able to register in the app with full name, email and password (all mandatory).
> The user can log into the app using email and password.
> If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
> When the user logs in, they are presented with the categories page.
> For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
> When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
> There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
> For a given category, the list of transactions is presented, ordered by the most recent.
> At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
> There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
> When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
> The user fills out a form to create a new category, indicating their name and icon (both mandatory).
> The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
> When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

>  ###The user fills out a form to create a new transaction with:_
> name (mandatory)
> amount (mandatory)
> categories (mandatory at least one)
> The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
> When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

**Testing requirements**

- Create unit and integration tests for all the most important components of your RoR application.

**Technical requirements**

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

### CODE REVIEW

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## Hi @Theophileaseh👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter check fails, Please make sure to fix all linter errors and go for another review. Good luck 👍

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
```
Comment (t-yanick) [inline README.md:68]: - [ ] As it is  a [requirement](https://gitlab.com/microverse/guides/tse/code_review/code_review_guidelines_2_0/-/blob/master/ror/capstone/ror_capstone.md) for this project, " `The Creative Commons license of the design requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.` ", kindly give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Budget App
+
+> An application built using Ruby on Rails to manage the expenses of any user. It allows the user the create categories under which are various transactions for the category. 
+
+## Screenshot
+| Categories Page |   Transactions page |
+| ---------- | ----------- |
+ | ![CategoriesPage](img/1.png) | ![TransactionsPage](img/2.png) |
+</p>
+
```
Comment (t-yanick) [inline README.md:13]: - [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) for this project, " `The project should be deployed and accessible online.` " , please kindly deploy the application using Heroku, Netlify or GitHub Pages.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @thecodechaser, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: Correct use of Gitflow. 
:heavy_check_mark: Descriptive `PR` summary,
:heavy_check_mark: Clean/Organized code,
:heavy_check_mark: Good commit messages,
:heavy_check_mark: Descriptive Readme file.
:heavy_check_mark: Great job with test coverage

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

- [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) for this project:
#### Project documentation
Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:

- a description of the project.
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.
- For recording the video you can use tools like [Loom](https://www.loom.com/) that let you share a private link to the recording, and configure a shot that shows your computer screen and your face at the same time in a small picture. Please kindly record a video presenting your project. 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (muneebulrehman) [reply README.md:13]: @Theophileaseh, as requested by the previous reviewer, please make the application online accessible by deploying it on any platform you like as it is a part of the requirement. Kindly do so and then submit for a re-review.

File: README.md (hunk 1 above)
Comment (muneebulrehman) [reply README.md:68]: As requested by the previous reviewer, kindly give the credit to the author of the design in the Readme file. Kindly do so and then submit for a re-review.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @Theophileaseh , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights
- No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      <%= button_to "Logout", destroy_user_session_path, method: :delete, data: { turbo: false }, class: "nav-login" %>
+      </div>
+        <% else %>
+        <div><%= button_to "Log in", new_user_session_path, class: "nav-login" %> </div>
+      <% end %>
+    </div>
+    </div>
+
+    <div class="notification-area">
+    <p class="notice"><%= notice %></p>
```
Comment (phelian23) [inline app/views/layouts/application.html.erb]: - [ ] As requested in the project requirements, the app should have a splash screen (a simple page with the name of the app with links to the sign up and log in pages). Kindly implement this.
figma

The splash screen should be the root page when the user is not logged in

File: config/database.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+#     url: <%= ENV["MY_APP_DATABASE_URL"] %>
+#
+# Read https://guides.rubyonrails.org/configuring.html#configuring-a-database
+# for a full overview on how database connection configuration can be specified.
+#
+production:
+  <<: *default
+  database: Budget_app_production
+  username: postgres
+  password: <%= ENV["BUDGET_APP_DATABASE_PASSWORD"] %>
```
Comment (phelian23) [inline config/database.yml:90]: - [ ] while I'm trying to interact with the app, I get this error page
Capture

Most times it is caused by not migrating the database to heroku after pushing to heroku. Kindly ensure that you migrates the database to heroku, and the apps works as it should.

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @Theophileaseh,

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Top highlight 🎯
The Pull Request is following the GitHub Flow ✅
The PR title and description are professional and descriptive ✅
The commit messages are meaningful ✅
The README is well documented ✅
Linters are working properly ✅

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @Theophileaseh 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**